core/src/test/java/com/expediagroup/streamplatform/streamregistry/core/services/StreamServiceTest.java

```diff
@@ -182,6 +183,35 @@ public void update() {
     verify(streamRepository).saveSpecification(entity);
   }
 
+  @Test(expected = ValidationException.class)
+  public void updateWithChangedSchemaKey() {
+    StreamKey key = new StreamKey();
+    key.setDomain("domain");
+    key.setName("stream");
+    key.setVersion(1);
+    Stream existingEntity = new Stream(key, new Specification(), new SchemaKey("domain", "stream_v1"));
+    Stream updatedEntity = new Stream(key, new Specification(), new SchemaKey("domain", "stream_v2"));
+    when(streamRepository.findById(key)).thenReturn(Optional.of(existingEntity));
+    streamService.update(updatedEntity);
```

Comment:
Should this throw an error?

Reply:
yeah i have added this 
`@Test(expected = ValidationException.class)`
do you want me to add error message ??

Reply:
now asserting the exact error .